runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngineConfig.kt

```diff
@@ -19,6 +24,10 @@ public class OkHttpEngineConfig private constructor(builder: Builder) : HttpClie
     }
 
     public class Builder : HttpClientEngineConfig.Builder() {
+        /**
+         * The maximum number of connections to open to a single host. Defaults to 5.
+         */
+        public var maxConnectionsPerHost: UInt = 5u
```

Comment:
Missed this in my review...

Question/Correctness: Should this match `maxConnections` by default?

Reply:
It could. I have no feeling for sensible defaults here and so just picked the value that was previously in effect from the `Dispatcher` anyway.

Reply:
I think it makes sense as the default (`maxConnections` that is) because when users aren't sharing an HTTP engine then the setting actually makes sense (most services will use a single host with S3 being a notable exception). 

Otherwise if a user configures `maxConnections` but not `maxConnectionsPerHost` they will actually be limited by the latter perhaps without realizing it. I think `maxConnectionsPerHost` probably makes the most sense to tune when you are sharing an engine across service clients.

Thoughts?

Reply:
That makes a lot of sense actually. I'll update this to a nullable field in the builder and default it to `maxConnections` in the config initializer.